fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java

```diff
@@ -111,14 +115,26 @@ private Map<String, String> requestHeaders(HttpServletRequest request) {
     if (request.getHeaderNames() != null) {
       for (Enumeration<String> h = request.getHeaderNames(); h.hasMoreElements(); ) {
         String headerName = h.nextElement();
-        String headerValue = request.getHeader(headerName);
+        String headerValue = getRedactedHeaderValue(headerName, request.getHeader(headerName));
         headers.put(headerName, headerValue);
       }
     }
 
     return headers;
   }
 
+  private String getRedactedHeaderValue(String headerName, String headerValue) {
+    if (headerName.startsWith(SECRET_DATA_HEADER_FORMAT)) {
+      return SECRET_DATA_VALUE;
+    } else {
+      if (headerName.contains(AUTHORIZATION_HEADER)) {
+        return headerValue.trim().split("\n")[0].split(" ")[0] + " " + SECRET_DATA_VALUE;
```

Comment:
This is technically correct, but we can make it slightly easier to read.  You can extract new private method `removeCredentialsFromAuthorizationHeader` or something like that. It will be instantly obvious what this string transformation does.